### PR TITLE
SDK-1279: Depricated config path value resolver, changed task definit…

### PR DESCRIPTION
**Pull request by vol4onok**
…ion, adjusted path errors and path question. Fixed and wrote tests

## Issue ticket link
Ticket: https://spryker.atlassian.net/browse/SDK-1279.

## Checklist before requesting a review
- [ ] I have added ticket link to the PR description.
- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

### CODE REVIEW

**Review by sergeyspryker — CHANGES_REQUESTED** [review]

File: src/Extension/ValueResolver/PriorityPathValueResolver.php [hunk 1]
```diff
@@ -38,10 +38,17 @@ public function getId(): string
...
+        }
+
         $relativePath = (string)parent::getValue($context, $settingValues, $optional);
 
-        if (!$this->getSettingPaths()) {
-            throw new InvalidConfigurationException(sprintf('`%s` resolver doesn\'t have any paths', $this->getId()));
+        if (strpos($relativePath, DIRECTORY_SEPARATOR) === 0) {
+            throw new UnresolvableValueExceptionException('Absolute path is forbidden due to security reasons.');
+        }
+        if (strpos($relativePath, '..' . DIRECTORY_SEPARATOR) === 0) {
```
[the remark this answers is not in the document]
Comment (vol4onok) [reply src/Extension/ValueResolver/PriorityPathValueResolver.php:50]: ok

**Review by sergeyspryker — DISMISSED** [review]

File: src/Infrastructure/Service/ErrorCommandListener.php [hunk 2]
```diff
@@ -9,33 +9,45 @@
...
-            new ProjectWorkflowException(
-                'You need to init or update the SDK. you need to run \'sdk:init:sdk\' or \'sdk:update:all\' command.',
-                0,
-                $event->getError(),
-            ),
-        );
         $event->setExitCode(0);
+
+        if ($event->getError() instanceof TableNotFoundException || $event->getError() instanceof SettingsNotInitializedException) {
+            $event->getOutput()->writeln(sprintf('<error>You need to init or update the SDK. you need to run \'%s\' or \'%s\' command.</error>', InitSdkCommand::NAME, UpdateSdkCommand::NAME));
```
Comment (pavelmaksimov25) [inline src/Infrastructure/Service/ErrorCommandListener.php:46]: What if we decorate Symfony command output with additional functions like `writeErrorLn()`, `writeWarningLn()`, etc.?

File: src/Extension/ValueResolver/PriorityPathValueResolver.php [hunk 3]
```diff
@@ -53,12 +61,13 @@ public function getValue(ContextInterface $context, array $settingValues, bool $
...
             }
             $path = implode(DIRECTORY_SEPARATOR, [$path, $relativePath]);
+
             if (file_exists($path)) {
                 return $this->formatValue($path);
             }
         }
 
-        throw new UnresolvableValueExceptionException('Can\'t resolve path.');
+        throw new UnresolvableValueExceptionException('Invalid path provided.');
```
Comment (pavelmaksimov25) [inline src/Extension/ValueResolver/PriorityPathValueResolver.php:70]: Normally it is better to have success scenario at the end. For this reason and for the readability perspectives, I'd recommend extracting `foreach` loop above to the separate method `extractFormattedPathValue()` and do the following here instead
```php
$pathValue = $this->extractFormattedPathValue($settingValues);
if (!$pathValue) {
     throw new UnresolvableValueExceptionException('Invalid path provided.');
}

return $pathValue;

File: src/Infrastructure/Service/ErrorCommandListener.php (hunk 2 above)
Comment (vol4onok) [reply src/Infrastructure/Service/ErrorCommandListener.php:46]: I'm not sure that we need to invest time to it now

**Review by pavelmaksimov25 — DISMISSED** [review]

**Review by pavelmaksimov25 — APPROVED** [review]

**Pull request merged**